### PR TITLE
Fix classloader issue

### LINKED ISSUE #29988: Classloader issue when @Context injecting implementation provided by the application
**Describe the bug**  
The following exception can occur whenever an application provided implementation for an `@Context` injection is in a resource class:

`[11/6/24, 12:56:44:706 CST] 0000006d g.jboss.resteasy.core.providerfactory.DefaultExceptionMapper E RESTEASY002375: Error processing request GET /CustomSecurityContext/CustomSecurityContextResource/GetCustom - com.ibm.ws.jaxrs.fat.customsecuritycontext.CustomSecurityContextResource.requestCustomSecurityInfo
java.lang.IllegalArgumentException: com.ibm.ws.jaxrs.fat.customsecuritycontext.filter.WrapperIntf referenced from a method is not visible from class loader: org.eclipse.osgi.internal.loader.EquinoxClassLoader @501ed359
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.ensureVisible(Proxy.java:881)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.validateProxyInterfaces(Proxy.java:707)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:635)
	at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$1(Proxy.java:440)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:438)
	at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1034)
	at org.jboss.resteasy.core.ContextParameterInjector.createProxy(ContextParameterInjector.java:233)
	at org.jboss.resteasy.core.ContextParameterInjector.inject(ContextParameterInjector.java:108)
	at org.jboss.resteasy.core.PropertyInjectorImpl.inject(PropertyInjectorImpl.java:135)
	at org.jboss.resteasy.cdi.JaxrsInjectionTarget.inject(JaxrsInjectionTarget.java:71)
	at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:165)
	at org.jboss.weld.contexts.AbstractContext.get(AbstractContext.java:96)
	at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:104)
	at org.jboss.weld.bean.ContextualInstanceStrategy$CachingContextualInstanceStrategy.get(ContextualInstanceStrategy.java:182)
	at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
	at org.jboss.weld.bean.proxy.ContextBeanInstance.getInstance(ContextBeanInstance.java:101)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.getInstance(ProxyMethodHandler.java:136)
	at com.ibm.ws.jaxrs.fat.customsecuritycontext.CustomSecurityContextResource$Proxy$_$$_WeldClientProxy.requestCustomSecurityInfo(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:397)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:356)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:70)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:397)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:245)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:723)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1266)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:754)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:451)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:197)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:100)
	at com.ibm.ws.security.jaspi.JaspiServletFilter.doFilter(JaspiServletFilter.java:58)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:203)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:93)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1069)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1260)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5096)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:328)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1047)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1260)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:476)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:435)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
`

**Steps to Reproduce**  
Create any RESTful Web Services application containing a `@Context` injection where the implementation class is contained within the application .jar.   For example:

Implementation class:
```
...
import javax.ws.rs.core.SecurityContext;
public class CustomSecurityContextWrapper implements SecurityContext, WrapperIntf {


```
Resource class:
```
...
import javax.ws.rs.core.SecurityContext;
...
@Path("CustomSecurityContextResource")
@RequestScoped
public class CustomSecurityContextResource {

    @Context
    SecurityContext securityContext;

    @GET
    @Path("Get")
    @RolesAllowed("admin")
    public Response requestSecurityInfo() {
```

**Expected behavior**  
It is expected that the `@Context` injection will work with no errors.

**Diagnostic information:**  
 - OpenLiberty Version: 21.0.0.12 - 24.0.0.11
 - Affected feature(s): restfulWS-3.0, restfulWS-3.1

### PR DESCRIPTION
**Pull request by jim-krueger**
Fixes #29988
Fixes #PH63904
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0_fat,com.ibm.ws.jaxrs.2.0.cdi.1.2_fat,com.ibm.ws.jaxrs.2.0.client_fat,com.ibm.ws.jaxrs.2.1_fat,com.ibm.ws.jaxrs.2.1_fat_executor,com.ibm.ws.jaxrs.2.1_fat_extended,com.ibm.ws.jaxrs.2.1.cdi.2.0_fat,com.ibm.ws.jaxrs.2.1.client_fat,com.ibm.ws.jaxrs.2.1.sse_fat,com.ibm.ws.jaxrs.2.x_fat_clientProps,io.openliberty.restfulWS.3.0_fat,io.openliberty.restfulWS.3.0.cdi.3.0_fat,io.openliberty.restfulWS.3.0.client_fat,io.openliberty.restfulWS.3.1_fat

**Comment by jim-krueger** [comment]
#libby

**Review by WhiteCat22 — DISMISSED** [review]

File: dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/customsecuritycontext/src/com/ibm/ws/jaxrs/fat/customsecuritycontext/CustomSecurityContextParamResource.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2024 IBM Corporation and others.
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/customsecuritycontext/src/com/ibm/ws/jaxrs/fat/customsecuritycontext/CustomSecurityContextParamResource.java]:
```suggestion
 * Copyright (c) 2024 IBM Corporation and others.
```

**Review by jim-krueger — COMMENTED** [review]
fix copyright

**Review by jhanders34 — APPROVED** [review]

**Review by WhiteCat22 — APPROVED** [review]

**Comment by jim-krueger** [comment]
#libby

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=d4cb6289-220e-4ee2-9400-6f434e09f3c3

Target locations of links might be accessible only to IBM employees.

**Comment by jim-krueger** [comment]
!libby

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=cf10805a-3dd0-4437-8c9b-18bac78b249d

Target locations of links might be accessible only to IBM employees.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=d4cb6289-220e-4ee2-9400-6f434e09f3c3 completed successfully!

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=cf10805a-3dd0-4437-8c9b-18bac78b249d completed successfully!

**Pull request merged**